### PR TITLE
Improve RegexUtils - Handle Edge Cases

### PR DESCRIPTION
**Pull request by ritishkumarsingh**
**Goal**
• Prototype a bounded private executor for `RegexUtils` to cap concurrent regex work.

**Outcome**  
• Tests didn’t surface a practical benefit over the existing `managedBlock` + common-pool design.  
• Added static pool and API churn felt heavier than the gain.

**Action**  
Branch reset to match master and experiment parked for now.

### CODE REVIEW

File: connect-utils/src/main/java/com/github/jcustenborder/kafka/connect/utils/RegexUtils.java [hunk 1]
```diff
@@ -21,6 +21,11 @@
 import java.util.concurrent.CompletionException;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicBoolean;
```
[the remark this answers is not in the document]
Comment (ritishkumarsingh) [reply connect-utils/src/main/java/com/github/jcustenborder/kafka/connect/utils/RegexUtils.java:28]: AtomicBoolean is used in RegexExecutor and is imported at the top (java.util.concurrent.atomic.AtomicBoolean). No change required.

File: connect-utils/src/main/java/com/github/jcustenborder/kafka/connect/utils/RegexUtils.java [hunk 2]
```diff
@@ -58,21 +63,44 @@ private RegexUtils() {
...
-      result = operation.get();
+      if (done.compareAndSet(false, true)) {
+        result = operation.get();
+      }
       return true;
     }
 
     public boolean isReleasable() {
-      return false;
+      return done.get();
```
[the remark this answers is not in the document]
Comment (ritishkumarsingh) [reply connect-utils/src/main/java/com/github/jcustenborder/kafka/connect/utils/RegexUtils.java:103]: It already does: our method returns true whether the regex ran (compareAndSet won) or was skipped. The implementation fully satisfies the ManagedBlocker contract, so no further adjustment is needed.

File: connect-utils/src/main/java/com/github/jcustenborder/kafka/connect/utils/RegexUtils.java [hunk 3]
```diff
@@ -54,25 +62,69 @@
...
   private RegexUtils() {
     // Prevent instantiation
   }
 
+  // Dedicated daemon-thread pool to isolate regex execution from the common ForkJoinPool.
+  private static final int MAX_REGEX_THREADS = Math.max(4, Runtime.getRuntime().availableProcessors() * 2);
+  private static final int QUEUE_MULTIPLIER = 100;
+
+  private static final ExecutorService REGEX_EXECUTOR_SERVICE = new ThreadPoolExecutor(
+      MAX_REGEX_THREADS,               // corePoolSize = max. Core threads never time out (unless allowCoreThreadTimeOut is enabled), so pool size is effectively fixed.
```
[the remark this answers is not in the document]
Comment (ritishkumarsingh) [reply connect-utils/src/main/java/com/github/jcustenborder/kafka/connect/utils/RegexUtils.java:75]: “Core threads never time out (unless allowCoreThreadTimeOut is enabled)”.
That reflects the actual behaviour, so nothing to change.

File: connect-utils/src/main/java/com/github/jcustenborder/kafka/connect/utils/RegexUtils.java [hunk 4]
```diff
@@ -141,14 +201,18 @@ public static String replaceAll(
    * @param input     The input string
    * @param timeoutMs The timeout in milliseconds
    * @return true if the pattern is found, false otherwise
-   * @throws InterruptedException if the current thread is interrupted
-   * @throws ExecutionException   if the operation throws an exception
-   * @throws TimeoutException     if the operation exceeds the specified timeout
+   * @throws InterruptedException       if the current thread is interrupted
+   * @throws ExecutionException         if the operation throws an exception
+   * @throws TimeoutException           if the operation exceeds the specified timeout
+   * @throws RejectedExecutionException if the internal pool is saturated
```
[the remark this answers is not in the document]
Comment (ritishkumarsingh) [reply connect-utils/src/main/java/com/github/jcustenborder/kafka/connect/utils/RegexUtils.java:207]: The exception is mentioned only in Javadoc; compilation doesn’t require an import.
